Add support for bool datatype (#601)

riskybacon opened the pull request:
This seems like a possibly naive fix for #601. This unblocked us when using pyprof, but it's not clear to me if it breaks other use cases.

mcarilli commented:
@mkolod @adityaiitb as pyprof authors, I don't know anything about the internals

ptrblck commented on apex/pyprof/prof/utility.py:
```diff
@@ -1,4 +1,5 @@
 from functools import reduce
+import pdb
```
Is the `pdb` import necessary here or could it be removed?

riskybacon replied:
Sorry about that, thanks for the catch, I'll remove it.

ptrblck commented:
@riskybacon Could you add a test case please, so that we can make sure it'll work for this new `dtype`?

riskybacon commented:
I've added unit tests for both a bool and non-bool test case, but I'm not too happy with how I added the test.

To test the code, I exposed `data` and `prof` from `apex/pyprof/__init__.py` and `apex/pyprof/__init__.py`. This doesn't feel right to me, but it seemed like the only way to expose those functions to a unit test.

Also, the test data inside of `TestPyProfData` seems like a brittle solution.

I'm open to suggestions :)
